crates/proto/src/domain/blocks.rs

```diff
@@ -1,14 +1,14 @@
-use miden_objects::BlockHeader;
+use miden_objects::{crypto::merkle::MerklePath, BlockHeader};
 
 use crate::{
     errors::{ConversionError, MissingFieldHelper},
-    generated::block_header,
+    generated::block,
```

Comment:
It is a bit difficult to tell what's what by looking at `block::BlockHeader` types below. I would probably alias this as:
```rust
generated::block as proto
```
And then everywhere below we could say just `proto::BlockHeader`.